File: cpp/grains/grains.cpp
#include "grains.h"

namespace grains {

    typedef uint_fast64_t grains_count;

    grains_count square(int n) {
        return grains_count(1) << (n-1);
    }

    grains_count total() {
        return ~0; // 2^64-1
    }

}

File: go/grains/grains.go
package grains

import (
	"errors"
)

func Square(n int) (uint64, error) {
	if n < 1 || n > 64 {
		return 0, errors.New("Bad")
	}
	return uint64(1 << uint(n-1)), nil
}

func Total() uint64 {
	return (1 << 64) - 1
}
